Source/WebCore/style/values/anchor-position/StylePositionTryFallback.cpp

```diff
@@ -0,0 +1,188 @@
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "StylePositionTryFallback.h"
+
+#include "StyleBuilderChecking.h"
+#include "StylePrimitiveKeyword+CSSValueConversion.h"
+#include "StylePrimitiveKeyword+CSSValueCreation.h"
+#include "StylePrimitiveKeyword+Logging.h"
+#include "StylePrimitiveKeyword+Serialization.h"
+#include "StylePropertiesInlines.h"
+
+namespace WebCore {
+namespace Style {
+
+bool PositionTryFallback::PositionArea::operator==(const PositionTryFallback::PositionArea& other) const
+{
+    if (properties && other.properties) {
+        if (properties == other.properties)
+            return true;
+
+        Ref strongProperties = *properties;
+        Ref strongOtherProperties = *other.properties;
+
+        auto lhsPositionArea = strongProperties->getPropertyCSSValue(CSSPropertyPositionArea);
+        ASSERT(lhsPositionArea);
+
+        auto rhsPositionArea = strongOtherProperties->getPropertyCSSValue(CSSPropertyPositionArea);
+        ASSERT(rhsPositionArea);
+
+        return *lhsPositionArea == *rhsPositionArea;
+    }
+
+    return !properties && !other.properties;
+}
+
+bool PositionTryFallback::operator==(const PositionTryFallback& other) const
+{
+    if (positionArea.properties && other.positionArea.properties)
+        return positionArea == other.positionArea;
+
+    if (!positionArea.properties && !positionArea.properties)
+        return ruleAndTactics == other.ruleAndTactics;
+
+    // If we got here, this and other don't have the same type (e.g comparing position-area with rule + tactics)
+    return false;
+}
+
+// MARK: - Conversion
+
+auto CSSValueConversion<PositionTryFallback>::operator()(BuilderState& state, const CSSValue& value) -> PositionTryFallback
+{
+    if (RefPtr valueList = dynamicDowncast<CSSValueList>(value)) {
+        if (valueList->separator() != CSSValueList::SpaceSeparator) {
+            state.setCurrentPropertyInvalidAtComputedValueTime();
+            return { };
+        }
+
+        auto rule = std::optional<ScopedName> { };
+        auto tactics = SpaceSeparatedVector<PositionTryFallbackTactic> { };
```

Comment:
Would this be better style? Which do you prefer?
```suggestion
        std::optional<ScopedName> rule;
        SpaceSeparatedVector<PositionTryFallbackTactic> tactics;
```

Reply:
I prefer the `auto foo =` one, as seeing the variable name seems more important to me.